sdk/core/System.ClientModel/src/Convenience/ClientResultException.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.ClientModel.Primitives;
+using System.Globalization;
+using System.IO;
+using System.Runtime.Serialization;
+using System.Text;
+
+namespace System.ClientModel;
+
+[Serializable]
+public class ClientResultException : Exception, ISerializable
+{
+    private const string DefaultMessage = "Service request failed.";
+
+    private readonly PipelineResponse? _response;
+    private int _status;
```

Comment:
@KrzysztofCwalina: why do we need both the response and status? PipelineResponse already has status.
https://github.com/Azure/azure-sdk-for-net/pull/41009#discussion_r1442147914

Reply:
We don't always have a PipelineResponse - the Azure.Core semantics are that if there is no raw response, status returns zero, and I wanted to initialize this value in the constructor.  Let me see if there is another way, though.

Reply:
We need this to support back-compatibility with RequestFailedException -- specifically, RFE has some constructors that allow setting the status, and so we need a backing-store for that since we may not have a PipelineResponse, and its status is read-only if we did.  [Here's a link to the implementation in the Azure.Core 2.0 feature branch](https://github.com/Azure/azure-sdk-for-net/blob/feature/core-experiment/sdk/core/Azure.Core/src/RequestFailedException.cs#L119) in case you're able to suggest a better approach!

Reply:
> Let me see if there is another way, though.

Can we remove the filed and have the following property implementation:

public virtual int Status => _response?.Status??0; // returns zero when response == null

Reply:
I don't see how that allows the RequestFailedException subtype in Azure.Core to set a custom value, but maybe I am missing something?

Reply:
The subtype could override the property and return a value of a field (that it set)

Reply:
I see.  So if it had its own custom field set it would get it from that, otherwise it would get it from the base type?  That should work - thanks!  It also removes the protected setter from the API, which I like better too.

Reply:
I found one more - the serialization constructor needs a way to set it?  Is there a way around that one?

Reply:
I'm not able to think of a way to address the serialization constructor without a backing field for the status code, unfortunately.  One thought I had is to create a dummy PipelineResponse to hold the status, but if someone called GetRawResponse it would be returned and I don't think that is desired -- I think we want null returned from that to indicate that no real response is set.

Reply:
Do we need to actually serialize the status? Is that useful? If yes, then we at least need to somewho make sure that the exception status and the PipelineResponse status are the same. Today, there does not seem to be any guarantee

Reply:
I think the BCL team doesn't think we should be serializing any of it :)
But, I remember Levi saying that if we do implement the serialization constructor, we should make sure it serializes status.

I could move it into the Azure.Core-type only and have it be non-functional on the ClientModel type, and we could wait for a customer to report an error?

Although @KrzysztofCwalina, can you share your specific concern about having one additional int field in an exception?  Aren't exceptions already a low-performance domain?

Reply:
Filed https://github.com/Azure/azure-sdk-for-net/issues/41081 to track.